dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/validation/hooks/RelationshipsValidationHook.java

```diff
@@ -73,21 +74,39 @@ public void validateRelationship( ValidationErrorReporter reporter, Relationship
             bundle.getPreheat().getAll( RelationshipType.class ) );
 
         // No need to check additional data if there are missing information on
-        // the
-        // Relationship
+        // the Relationship
         if ( isValid )
         {
             validateRelationshipLinkToOneEntity( reporter, relationship );
             validateRelationshipConstraint( reporter, relationship, bundle );
 
             validateAutoRelationship( reporter, relationship );
 
+            validateDuplication( reporter, relationship, bundle );
+
             validateReferences( reporter, relationship, relationship.getFrom() );
             validateReferences( reporter, relationship, relationship.getTo() );
         }
 
     }
 
+    private void validateDuplication( ValidationErrorReporter reporter, Relationship relationship,
+        TrackerBundle bundle )
+    {
+        org.hisp.dhis.relationship.Relationship relationshipUsingKey = bundle.getPreheat()
```

Comment:
name it `existingRelationship` like you did in the other hook

Reply:
I will actually rename it to duplicatedRelationship